python/pyspark/__init__.py

```diff
@@ -36,6 +36,31 @@
       Finer-grained cache persistence levels.
 
 """
+import os
+import sys
+
+import xml.etree.ElementTree as ET
+
+if (os.environ.get("SPARK_HOME", "not found") == "not found"):
+    raise ImportError("Environment variable SPARK_HOME is undefined.")
+
+spark_home = os.environ['SPARK_HOME']
+pom_xml_file_path = os.path.join(spark_home, 'pom.xml')
```

Comment:
There should be an version number in the path of assembly jar, could we use that?